common/ferrostar/src/simulation.rs

```diff
@@ -14,22 +15,20 @@ pub enum SimulationError {
     NotEnoughPoints,
 }
 
-#[derive(uniffi::Record, Clone)]
+#[derive(uniffi::Record, Clone, PartialEq)]
 #[cfg_attr(test, derive(Serialize))]
 pub struct LocationSimulationState {
     pub current_location: UserLocation,
     remaining_locations: Vec<GeographicCoordinate>,
 }
 
-#[derive(uniffi::Enum)]
-pub enum SimulationAdvanceStyle {
```

Comment:
`SimulationAdvanceStyle` was dropped because I determined that it wasn't logical to specify this at every iteration but rather to decide upfront whether to interpolate.